src/OrchardCore.Modules/OrchardCore.Users/Views/Account/LinkExternalLogin.cshtml

```diff
@@ -7,7 +7,7 @@
 <form asp-controller="Account" asp-action="LinkExternalLogin" asp-route-returnurl="@ViewData["ReturnUrl"]" method="post" class="form-horizontal no-multisubmit">
     <h4>@T["Link your account."]</h4>
     <p class="text-info">
-        @T["You've successfully authenticated with <strong>{0}</strong>. You already have an account with this email address. Enter your local account password and click the Register button to link the accounts and finish logging in.", ViewData["LoginProvider"]]
+        @T["You've successfully authenticated with <strong>{0}</strong>. You already have an account associated with <strong>{1}</strong>. Enter your local account password and click the Register button to link the accounts and finish logging in.", ViewData["LoginProvider"], ViewData["ExternalUserIdentifier"]]
```

Comment:
I don't think the identifier used to match the account is interesting for a user, or even more, something that they should be allowed to know and perhaps use maliciously. So, I'd remove `GetIdentifierKey()` and everything related to it, including the `ExternalUserIdentifier` field below, and change the message to this:

```
"You've successfully authenticated with <strong>{0}</strong>. You already have an account associated with this external login. Enter your local account password and click the Register button to link the accounts and finish logging in."
```

Reply:
I'm not completely agree with this change. The meaning of the code is to allow to link the external login data to an existing local account.

Probably:

```
"You've successfully authenticated with <strong>{0}</strong>. You already have an account associable with this external login informations. Enter your local account password and click the Register button to link the accounts and finish logging in."
```

Reply:
I don't really see why your version is better suited. Note that this is for end-users, including non-technical ones. Perhaps the message I've written should be further simplified too.

Reply:
is a sujective discussion... I don't want to "push" about it more 😄 ... I thinked is more correct to use 'associable' because the local account is not alrady associated to the external login data in the moment when user see this page. I have already released your version. 

Thank you

Reply:
Ah I see. Then "You already have an account that can be linked with this external login."?

Reply:
I think this was the default behavior, if a local account exists to link the external login